libhif/hif-context.c

```diff
@@ -151,6 +151,35 @@ hif_context_init(HifContext *context)
     priv->cache_age = 60 * 60 * 24 * 7; /* 1 week */
     priv->override_macros = g_hash_table_new_full(g_str_hash, g_str_equal,
                                                   g_free, g_free);
+
+    (void) hif_context_get_base_arch(context);
+}
+
+/**
+ * hif_context_globals_init:
+ * @error: Error
+ *
+ * Sadly RPM has process global data.  You should invoke
+ * this function early on in process startup.  If not,
+ * it will be invoked for you.
+ */
+gboolean
```

Comment:
Why *do* we use gboolean?  So we can return an error - the `rpmReadConfigFiles` can fail.